File: 8. Week 8 - Colour/exercise/r/visualization.R
# Visualize Monthly Distance with Color Scale using ggplot2

if (!require(ggplot2)) install.packages("ggplot2")
if (!require(readr)) install.packages("readr")
library(ggplot2)
library(readr)

# We need to make sure the path points to our data file
data_path <- "../../data/distance_per_month.csv"

if (file.exists(data_path)) {
  df <- read_csv(data_path, show_col_types = FALSE)
  
  # Plot
  # TODO: Create a bar chart that uses color to represent the distance intensity
  # Use geom_col(aes(fill = TotalDistance))
  # Experiment with scale_fill_gradient()
  
  # p <- ggplot(df, aes(x = Month, y = TotalDistance, fill = TotalDistance)) + ...

  
  # Uncomment to save
  # ggsave("colored_bar_chart.png", plot = p, width = 12, height = 6)
  # print("Chart saved to colored_bar_chart.png")

} else {
  print(paste("Error: File not found at", data_path))
}

File: 8. Week 8 - Colour/exercise/python/visualization.py
import pandas as pd
import matplotlib.pyplot as plt
import matplotlib.cm as cm
import matplotlib.colors as mcolors
import numpy as np
import os

def visualize():
    data_path = '../../data/distance_per_month.csv'
    
    if not os.path.exists(data_path):
        print(f"Data not found at {data_path}")
        return

    # TODO: Load data
    # df = ...
    
    # Setup Colors
    # TODO: Normalize distance values
    # norm = mcolors.Normalize(...)
    
    # TODO: Choose a colormap
    # cmap = ...
    
    # TODO: Create colors for each bar
    # colors = ...
    
    plt.figure(figsize=(12, 6))
    
    # TODO: Create bar chart using your custom colors
    # bars = plt.bar(..., color=colors)
    
    # TODO: Add title and labels
    # plt.title(...)
    # plt.xlabel(...)
    # plt.ylabel(...)
    plt.xticks(rotation=45)
    
    # TODO: Add a Colorbar
    # sm = cm.ScalarMappable(...)
    # sm.set_array([])
    # plt.colorbar(...)
    
    plt.tight_layout()
    # TODO: Save
    # plt.savefig(...)
    
    print("Chart saved to colored_bar_chart.png")
    plt.show()

if __name__ == "__main__":
    visualize()
